hedy.py

```diff
@@ -2613,20 +2621,107 @@ def get_keywords_for_language(language):
     return keywords
 
 
-PARSER_CACHE = {}
+def _get_parser_cache_directory():
+    # TODO we should maybe store this to .test-cache so we can
+    # re-use them in github actions caches for faster CI.
+    # We can do this after #4574 is merged.
+    return tempfile.gettempdir()
 
 
-def get_parser(level, lang="en", keep_all_tokens=False):
+def _save_parser_to_file(lark, pickle_file):
+    # Store the parser to a file, a bit hacky because it is not
+    # pickle-able out of the box
+    # See https://github.com/lark-parser/lark/issues/1348
+
+    # Note that if Lark ever implements the cache for Earley parser
+    # or if we switch to a LALR parser we don't need this hack anymore
+
+    full_path = os.path.join(_get_parser_cache_directory(), pickle_file)
+
+    # These attributes can not be pickled as they are a module
+    lark.parser.parser.lexer_conf.re_module = None
+    lark.parser.lexer_conf.re_module = None
+
+    try:
+        with atomic_write_file(full_path) as fp:
```

Comment:
I found this gem in the utils so I just force pushed that version instead of the earlier implementation :)

Reply:
Forgot that existed, but good call. I now seem to recall there were some issues on Windows for the original implementation.